crates/engine/src/memory_engine/mod.rs

```diff
@@ -53,6 +54,45 @@ impl MemoryEngine {
             inner: Arc::new(RwLock::new(db)),
         }
     }
+
+    /// Write an operation
+    #[inline]
+    fn write_op(
+        inner: &mut RwLockWriteGuard<'_, HashMap<String, MemoryTable>>,
+        op: WriteOperation<'_>,
+    ) -> Result<(), EngineError> {
+        match op {
+            WriteOperation::Put { table, key, value } => {
+                let table = inner
+                    .get_mut(table)
+                    .ok_or_else(|| EngineError::TableNotFound(table.to_owned()))?;
+                let _ignore = table.insert(key, value);
+            }
+            WriteOperation::Delete { table, key } => {
+                let table = inner
+                    .get_mut(table)
+                    .ok_or_else(|| EngineError::TableNotFound(table.to_owned()))?;
+                let _ignore = table.remove(key);
+            }
+            WriteOperation::DeleteRange { table, from, to } => {
+                let table = inner
+                    .get_mut(table)
+                    .ok_or_else(|| EngineError::TableNotFound(table.to_owned()))?;
+                table.retain(|key, _value| {
+                    let key_slice = key.as_slice();
+                    match key_slice.cmp(from) {
+                        Ordering::Less => true,
+                        Ordering::Equal => false,
+                        Ordering::Greater => match key_slice.cmp(to) {
+                            Ordering::Less => false,
+                            Ordering::Equal | Ordering::Greater => true,
+                        },
```

Comment:
nit. Better use `matches!(key_slice.cmp(to), Ordering::Equal | Ordering::Greater)` to replace.

Reply:
done